fastcrypto/src/encoding.rs

```diff
@@ -266,3 +267,22 @@ impl<'de, const N: usize> DeserializeAs<'de, [u8; N]> for Base58 {
         Ok(array)
     }
 }
+
+/// Bech32 encoding
+pub struct Bech32(String);
+
+impl Bech32 {
+    pub fn decode(s: &str, hrp: &str) -> Result<Vec<u8>, eyre::Report> {
```

Comment:
encode receives hrp as an input, why not return it here as an output and let the caller verify consistency?

Reply:
i went back and forth on this, https://github.com/MystenLabs/sui/pull/15415/files#diff-ccd99b30fee16b246ae935ee7b8c4d336530cc6962dcd611695d73f9493e8876R223 here is the usage in sui